plugins/pencilblue/services/salesforce/salesforce_strategy_service.js

```diff
@@ -74,6 +74,9 @@ module.exports = function(pb) {
 
         async getSalesforceLoginSettings(req) {
             try {
+                if(req.session && req.session.authentication && req.session.authentication.user && req.session.authentication.user.jobSeekerProfileId){
```

Comment:
i would suggest to use ```external_user_id``` instead of ```jobSeekerProfileId ``` to determine if user is logged in. otherwise, if user logged in and didn't create a profile, and then go to another site to login, he/she will encounter this error too. meanwhile, if the user is redirected to profile/view, it will redirect the user to profile/create directly.

Reply:
it would be better to put this kind of check as early as possible. Like at the beginning of `action render` instead of in the service.

Reply:
Updated please review again when you free